src/main/java/org/radarcns/management/service/mapper/decorator/SubjectMapperDecorator.java

```diff
@@ -39,12 +38,10 @@ public SubjectDTO subjectToSubjectDTO(Subject subject) {
         }
         SubjectDTO dto = delegate.subjectToSubjectDTO(subject);
         dto.setStatus(getSubjectStatus(subject));
-        Optional<Role> role = subject.getUser().getRoles().stream()
-                .filter(r -> r.getAuthority().getName().equals(AuthoritiesConstants.PARTICIPANT))
-                .findFirst();
-
-        role.ifPresent(role1 ->
-                dto.setProject(projectMapper.projectToProjectDTO(role1.getProject())));
+        Optional<Project> activeProject = subject.getActiveProject();
+        if (activeProject.isPresent()) {
```

Comment:
why not `ifPresent(project -> dto.setProject(projectMapper.projectToProjectDTO(project)))`?

Reply:
Pushed the changes. Thanks for the suggestions.